app/lib/admin/models.dart

```diff
@@ -43,15 +43,23 @@ class ModerationCase extends db.ExpandoModel<String> {
   @db.StringProperty(required: true)
   late String kind;
 
-  /// The package this notification or appeal concerns.
-  /// In the case of an appeal, reporter must be owner of said package.
+  /// The kind of the entity this notification or appeal concerns. On of:
+  /// `Package`, `PackageVersion` or `Publisher`.
   @db.StringProperty()
-  String? subjectPackage;
+  String? subjectKind;
 
-  /// The publisher this notification or appeal concerns.
-  /// In the case of an appeal, reporter must be a member of said publisher.
+  /// The fully qualified name of the entity this notification or appeal concerns.
+  /// - `Package:<package>`
+  /// - `PackageVersion:<package>/<version>`
+  /// - `Publisher:<publisherId>`
   @db.StringProperty()
-  String? subjectPublisher;
+  String? subjectFqn;
+
+  /// The local name of the entity (without the type qualifier) this notification or appeal concerns.
+  /// - `Package`: the package name
+  /// - `PackageVersion`: the `<package>/<version>`
+  /// - `Publisher`: the publisher ID
+  String? subjectLocalName;
```

Comment:
Curious, do we think this is better?

It'll mean that we need composite indexes? hmm, I guess `subjectFqn` will work.

hmm, I suppose this will make it easier to search for things...